### ISSUE TITLE
Build a replacement for GCM/FCM (used for Desktop notifications)

### ISSUE THREAD

**Issue by bsclifton** [brave/brave-browser#4431]
## Task
Chromium/Chrome have **GCM** ([Google Cloud Messaging](https://developers.google.com/cloud-messaging/) - deprecated April 2018) and **FCM** ([Firebase Cloud Messaging](https://firebase.google.com/docs/cloud-messaging/)) enabled. These protocols are used in delivering notifications on the Desktop product (at least, on Chromium/Chrome). Because these reach out to Google, we have disabled them:
- Disable GCM https://github.com/brave/brave-core/pull/714
- Disable FCM https://github.com/brave/brave-core/pull/908

Implementing this task would mean standing up a service (are there any off-the-shelf solutions?) that can handle routing these notifications in a private and secure way

Implementation of this task would unblock https://github.com/brave/brave-browser/issues/2362

## Associated Project
https://github.com/brave/brave-browser/issues/2362

Also see Slack conversation at https://bravesoftware.slack.com/archives/C7VLGSR55/p1557485850144300

**Comment by bsclifton** [comment]
cc: @rebron for prioritization

**Comment by WorldLanguages** [comment]
I honestly don't care if these services reach Google. Could there be some way push notifications can be enabled in Brave? A flag, something? Give users a chance to choose what to do until you implement an alternative to GCM/FCM, please.

**Comment by szaimen** [comment]
Any update on this?

**Comment by bsclifton** [comment]
Latest update:
- We should be able to get desktop notifications implemented without resolving this issue
- When resolving this issue, we might try to leverage the work [Mozilla](https://wiki.mozilla.org/Firefox/Push_Notifications) has done (including standing up our own server)

Will give an update in the other issue about where we're at 😄

**Comment by markmakes** [comment]
And here's how our entire development team now transitions off of Brave...

**Comment by szaimen** [comment]
@markmakes What?

**Comment by beautifwhale** [comment]
Also hitting this issue on Brave working with Googles cloud messaging. Any updates?

**Comment by LesterCovax** [comment]
Working for me now, at least with JoinApp (which was broken for a long time due to this).  More info here: https://github.com/brave/brave-browser/issues/2408#issuecomment-546843282

**Comment by aboydpoi** [comment]
Any update on this? @LesterCovax @bsclifton

**Comment by LesterCovax** [comment]
@aboydpoi What do you mean?  I said it was working for me. 😕

**Comment by bsclifton** [comment]
With the new setting, it should work great- no progress on a complete replacement for GCM/FCM. As shared above, [Mozilla does have an implementation](https://wiki.mozilla.org/Firefox/Push_Notifications) that we could explore, but we haven't prioritized this work (yet)

**Comment by totomaze** [comment]
> With the new setting, the feature itself should work great
> <img alt="Screen Shot 2019-12-30 at 10 03 22 AM" width="728" src="https://user-images.githubusercontent.com/4733304/71591988-aa3c9680-2aeb-11ea-8d78-453e88f87cfe.png">
> 
> No progress or updates on a complete replacement for GCM/FCM. As shared above, [Mozilla does have an implementation](https://wiki.mozilla.org/Firefox/Push_Notifications) that we could explore, but we haven't prioritized this work (yet)

Where is this option  in Version 0.70.123 Chromium: 78.0.3904.97 (Official Build) (64-bit) ?

**Comment by bsclifton** [comment]
@totomaze it's not there, as that is a fairly old build. Can you please update to a newer build? Current release channel is 1.1.23

**Comment by Fluf22** [comment]
Hey! 
Any news on the prioritization of this issue?
Thanks 😁

**Comment by LesterCovax** [comment]
I replied a while back that it works just fine now. I'd create a new issue
if you're having issues.

**Comment by shadow1349** [comment]
I'm not sure why, but blocking FCM has also broken Firebase Callable Functions in Angular. Here's a PR I made for it:

angular/angularfire#2475 (comment)

I don't think there is much the Angular team can do to fix it since it seems the issue is on Brave itself. Now my whole app is broken in Brave since I rely on callable functions.
